core/modules/installer/installer.authorize.inc

```diff
@@ -228,10 +232,30 @@ function installer_authorize_update_batch_finished($success, $results) {
       'type' => 'error',
     );
   }
-  // Since we're doing an update of existing code, always add a task for
-  // running update.php.
-  $results['tasks'][] = t('Your projects have been downloaded and updated.');
-  $results['tasks'][] = t('<a href="@update">Run database updates</a>', array('@update' => base_path() . 'core/update.php'));
+  $updates_needed = FALSE;
+  include_once(BACKDROP_ROOT . '/core/includes/install.inc');
+  foreach ($successful_projects as $module) {
+    module_load_install($module);
+    $updates = backdrop_get_schema_versions($module);
+    if ($updates !== FALSE) {
+      $current_schema = backdrop_get_installed_schema_version($module);
+      // If current schema is greater than 7000, bring it down to 999
+      $current_schema = $current_schema > 7000 ? 999 : $current_schema;
+      $latest_schema = end($updates);
+      if ($current_schema < $latest_schema) {
+        $updates_needed = TRUE;
+        break;
+      }
+    }
+  }
+  // Only add a task for running update.php if it is actually needed.
+  if ($updates_needed) {
+    $results['tasks'][] = t('<a href="@update">Run database updates</a>', array('@update' => base_path() . 'core/update.php'));
```

Comment:
Since the entire translatable string here is the link itself, perhaps use `l()` here?

Reply:
We also lost the `Your projects have been downloaded and updated.` message here. It should be added as a `backdrop_set_message()` of type `info` instead of a task though.